coding/cbor/encode.go

```diff
@@ -0,0 +1,75 @@
+package cbor
+
+import (
+	"io"
+
+	links "github.com/ipfs/go-ipld/links"
+	memory "github.com/ipfs/go-ipld/memory"
+	ma "github.com/jbenet/go-multiaddr"
+	cbor "github.com/whyrusleeping/cbor/go"
+)
+
+// Encode to CBOR, add the multicodec header
+func Encode(w io.Writer, node memory.Node, tags bool) error {
+	_, err := w.Write(Header)
+	if err != nil {
+		return err
+	}
+
+	return RawEncode(w, node, tags)
+}
+
+// Encode to CBOR, do not add the multicodec header
+func RawEncode(w io.Writer, node memory.Node, tags bool) error {
+	enc := cbor.NewEncoder(w)
+	if tags {
+		enc.SetFilter(func(val interface{}) interface{} {
```

Comment:
i believe this closure will generate a function each time. why this and not some simpler, more decomposed thing?